Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 282316)
+++ CMakeLists.txt	(working copy)
@@ -222,6 +222,12 @@
   add_definitions( -DCLANG_VENDOR="${CLANG_VENDOR} " )
 endif()
 
+set(CLANG_GRTE_ROOT "" CACHE STRING
+  "Root directory of GRTE runtime.")
+if( CLANG_GRTE_ROOT )
+  add_definitions( -DGRTE_ROOT="${CLANG_GRTE_ROOT}" )
+endif()
+
 set(CLANG_REPOSITORY_STRING "" CACHE STRING
   "Vendor-specific text for showing the repository the source is taken from.")
 
Index: lib/Driver/ToolChains.cpp
===================================================================
--- lib/Driver/ToolChains.cpp	(revision 282316)
+++ lib/Driver/ToolChains.cpp	(working copy)
@@ -4371,8 +4371,8 @@
 
   if (Distro == Exherbo && (Triple.getVendor() == llvm::Triple::UnknownVendor ||
                             Triple.getVendor() == llvm::Triple::PC))
-    return "/usr/" + Triple.str() + "/lib/" + Loader;
-  return "/" + LibDir + "/" + Loader;
+    return GRTE_ROOT "/usr/" + Triple.str() + "/lib/" + Loader;
+  return GRTE_ROOT "/" + LibDir + "/" + Loader;
 }
 
 void Linux::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
